cocos/core/timer.ts

```diff
@@ -0,0 +1,209 @@
+/*
+ Copyright (c) 2021 Xiamen Yaji Software Co., Ltd.
+
+ http://www.cocos.com
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated engine source code (the "Software"), a limited,
+  worldwide, royalty-free, non-assignable, revocable and non-exclusive license
+ to use Cocos Creator solely to develop games on your target platforms. You shall
+  not use Cocos Creator software for developing other software or tools that's
+  used for developing games. You are not granted to publish, distribute,
+  sublicense, and/or sell copies of Cocos Creator.
+
+ The software or tools in this License Agreement are licensed, not sold.
+ Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+*/
+/**
+ * @packageDocumentation
+ * @module core
+ */
+
+import { JSB, RUNTIME_BASED } from 'internal:constants';
+
+export interface TickByTimer {
+    tick: (deltaTime: number) => void;
+}
+
+export class Timer {
+    /**
+     * @en Expected frame rate of the timer.
+     * @zh 计时器的设定帧率。
+     */
+    public get frameRate () {
+        return this._frameRate;
+    }
+    public set frameRate (frameRate: number | string) {
+        if (typeof frameRate !== 'number') {
+            frameRate = parseInt(frameRate, 10);
+            if (Number.isNaN(frameRate)) {
+                frameRate = 60;
+            }
+        }
+        this._frameRate = frameRate;
+        this.frameTime = 1000 / frameRate;
+        this._setAnimFrame();
+    }
+
+    /**
+     * @en The delta time since last frame, unit: s.
+     * @zh 获取上一帧的增量时间，以秒为单位。
+     */
+    public get deltaTime () {
+        return this._deltaTime;
+    }
+
+    /**
+     * @en The total passed time since game start, unit: ms
+     * @zh 获取从游戏开始到现在总共经过的时间，以毫秒为单位
+     */
+    public get totalTime () {
+        return performance.now() - this._initTime;
+    }
+
+    /**
+     * @en The start time of the current frame.
+     * @zh 获取当前帧开始的时间。
+     */
+    public get frameStartTime () {
+        return this._startTime;
+    }
+
+    /**
+     * @en Check whether the timer is paused.
+     * @zh 判断计时器是否暂停。
+     */
+    public get paused () {
+        return this._paused;
+    }
+
+    public declare frameTime: number;
+
+    private declare _sys: TickByTimer;
```

Comment:
the name of `_sys` is kind of strange, I thought it was define in `sys.ts` at the first sight.
Maybe it should be named as `_ticker: Ticker` ?